scan-command/src/main/java/io/ballerina/scan/utils/ScanUtils.java

```diff
@@ -276,6 +301,242 @@ private static void unzipReportResources(InputStream source, File target) throws
+                TomlDocument ballerinaTomlDocument = ballerinaToml.tomlDocument();
+                Toml ballerinaTomlDocumentContent = ballerinaTomlDocument.toml();
+                Toml scanTable = ballerinaTomlDocumentContent.getTable(SCAN_TABLE).orElse(null);
+
+                if (scanTable != null) {
+                    TomlValueNode configPath = scanTable.get(SCAN_FILE_FIELD).orElse(null);
+
+                    if (configPath != null) {
+                        String scanTomlPath = configPath.toNativeValue().toString();
+                        Optional<Path> path = convertConfigurationPath(scanTomlPath);
+                        Optional<Path> absPath = path.isEmpty() || path.get().isAbsolute() ? path
+                                : Optional.of(project.sourceRoot().resolve(scanTomlPath));
+
+                        if (isLocalFile(absPath)) {
+                            return loadScanFile(targetDir, absPath.get(), outputStream);
+                        } else {
+                            try {
+                                URL url = new URL(scanTomlPath);
+                                return loadRemoteScanFile(targetDir, url, outputStream);
+                            } catch (MalformedURLException ex) {
+                                throw new RuntimeException("Failed to retrieve remote configuration file " +
+                                        "with exception: " + ex.getMessage());
+                            }
+                        }
+                    }
+                    outputStream.println("configPath for Scan.toml is missing!");
+                    return new ScanTomlFile();
+                }
+                Path scanTomlFilePath = Path.of(System.getProperty(USER_DIR_PROPERTY)).resolve(SCAN_FILE);
+
+                if (Files.exists(scanTomlFilePath)) {
+                    outputStream.println("Loading scan tool configurations from " + scanTomlFilePath + "...");
+                    return loadScanFile(targetDir, scanTomlFilePath, outputStream);
+                }
+                return new ScanTomlFile();
+            }
+            return new ScanTomlFile();
+        }
+        return new ScanTomlFile();
+    }
+
+    /**
+     * Returns the {@link Optional<Path> } if the path is valid.
+     *
+     * @param path the path to check
+     * @return the optional path of the configuration file
+     */
+    private static Optional<Path> convertConfigurationPath(String path) {
+        try {
+            return Optional.of(Paths.get(path));
+        } catch (InvalidPathException ex) {
+            return Optional.empty();
+        }
+    }
+
+    /**
+     * Checks if the provided file is a local file.
+     *
+     * @param path the path of the file
+     * @return true if the file is a local file, false otherwise
+     */
+    private static boolean isLocalFile(Optional<Path> path) throws InvalidPathException {
+        return path.isPresent() && new File(path.get().toString()).exists();
```

Comment:
```suggestion
        return path.isPresent() && path.toFile().isFile()
```

This will also check if this is a file (your code will be true even for a directory)